src/rgw/rgw_op.cc

```diff
@@ -308,7 +308,8 @@ static int read_policy(RGWRados *store,
     obj.init_ns(bucket, oid, mp_ns);
     obj.set_in_extra_data(true);
   } else {
-    obj = rgw_obj(bucket, object);
+    obj = rgw_obj(bucket, object.name);
+    obj.set_instance(s->object.instance);
```

Comment:
@oritwas reading the code further, I now understand the issue. We should get this fix in, and then rework rgw_obj and rgw_obj_key for future versions.